flatland/envs/agent_chains.py

```diff
@@ -12,31 +41,24 @@ class MotionCheck(object):
     """
 
     def __init__(self):
-        self.G = nx.DiGraph()
+        self.G = nx.DiGraph()  # nodes of type `Cell`
+        # TODO do we need the reversed graph at all?
```

Comment:
@hagrid67 @aiAdrian why do we need the reversed graph at all? Can we not just use the predecessors of directed graphs? Or does this have performance drawbacks?

Reply:
@hagrid67 thought of:
```
g = nx.DiGraph([ (0,1), (1,2), (2,3), (3,4), (5,6), (6,2) ])
print(list(g.predecessors(4)))
# returns: [3]
```